src/Estimators/src/FloatingBaseEstimator.cpp

```diff
@@ -604,3 +637,73 @@ bool FloatingBaseEstimator::setupFixedVectorParamPrivate(const std::string& para
     return true;
 }
 
+bool FloatingBaseEstimator::updateBaseStateFromIMUState(const FloatingBaseEstimators::InternalState& state,
+                                                        const FloatingBaseEstimators::Measurements& meas,
+                                                        iDynTree::Transform& basePose,
+                                                        iDynTree::Twist& baseTwist)
+{
+    iDynTree::Position imuPosition;
+    iDynTree::toEigen(imuPosition) = state.imuPosition;
+    iDynTree::Rotation imuRotation;
+    iDynTree::toEigen(imuRotation) = state.imuOrientation.toRotationMatrix();
+    auto A_H_IMU = iDynTree::Transform(imuRotation, imuPosition);
+
+    iDynTree::Twist v_IMU;
+    iDynTree::Vector3 imuLinVel, imuAngVel;
+    iDynTree::toEigen(imuLinVel) = state.imuLinearVelocity;
+    iDynTree::toEigen(imuAngVel) = state.imuOrientation.toRotationMatrix()*(meas.gyro - state.gyroscopeBias);
+    v_IMU.setLinearVec3(imuLinVel);
+    v_IMU.setAngularVec3(imuAngVel);
+
+    if (!m_modelComp.getBaseStateFromIMUState(A_H_IMU, v_IMU, basePose, baseTwist))
+    {
+        return false;
+    }
+    return true;
+}
+
+bool FloatingBaseEstimator::resetEstimator(const Eigen::Quaterniond& newBaseOrientation,
+                                           const Eigen::Vector3d& newBasePosition)
+{
+    iDynTree::Rotation Rb;
+    iDynTree::toEigen(Rb) = newBaseOrientation.toRotationMatrix();
+
+    iDynTree::Position pb;
+    iDynTree::toEigen(pb) = newBasePosition;
+
+    auto A_H_B = iDynTree::Transform(Rb, pb);
+    iDynTree::Transform IMU_H_RF, IMU_H_LF;
+    iDynTree::JointPosDoubleArray s(m_meas.encoders.size());
+    toEigen(s) = m_meas.encoders;
+
+    if (!m_modelComp.getIMU_H_feet(s, IMU_H_RF, IMU_H_LF))
+    {
+        std::cerr << "[FloatingBaseEstimator::resetEstimator] Could not reset estimator using new base pose." << std::endl;
+        return false;
+    }
+
+    auto A_H_IMU = A_H_B*m_modelComp.base_H_IMU();;
+    auto A_H_RF = A_H_IMU*IMU_H_RF;
+    auto A_H_LF = A_H_IMU*IMU_H_LF;
+
+    // convert to iDynTree Rotation to  Eigen angle axis and then to quaternion for consistent conversion
+    auto imuAngleAxis = Eigen::AngleAxisd(iDynTree::toEigen(A_H_IMU.getRotation()));
+    auto rfAngleAxis = Eigen::AngleAxisd(iDynTree::toEigen(A_H_RF.getRotation()));
+    auto lfAngleAxis = Eigen::AngleAxisd(iDynTree::toEigen(A_H_LF.getRotation()));
```

Comment:
I would avoid using `auto` with `Eigen`

Reply:
Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/85/commits/415e1729f4d047cfa520ce00633f7ede70796167